components/script/dom/document.rs

```diff
@@ -3019,6 +3033,52 @@ impl DocumentMethods for Document {
         elements
     }
 
+    // https://html.spec.whatwg.org/multipage/#dom-document-write
+    fn Write(&self, text: Vec<DOMString>) -> ErrorResult {
+        if !self.is_html_document() {
+            // Step 1.
+            return Err(Error::InvalidState);
+        }
+
+        // Step 2.
+        // TODO: handle throw-on-dynamic-markup-insertion counter.
+
+        if !self.is_active() {
+            // Step 3.
+            return Ok(());
+        }
+
+        let parser = self.get_current_parser();
+        let parser = match parser.as_ref() {
+            Some(parser) if parser.script_nesting_level() > 0 => parser,
```

Comment:
So the spec's idea of an insertion point corresponds to an active parser with a script nesting level > 0 in Servo? Is there some way we can document that better?